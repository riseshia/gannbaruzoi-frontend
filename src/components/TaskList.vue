<template>
  <div class="task-list">
    <h2>Task List</h2>
    <input
      type="text"
      :value="newTaskDescription"
      placeholder="Type new task!"
      @input="updateNewTaskDescription"
      @keyup.enter="createTask">
    <ul>
      <task-list-item
        v-for="edge in tasks"
        :key="edge.cursor"
        v-bind="edge.node"
        :logged-size="edge.node.logs.length"
      />
    </ul>
  </div>
</template>

<script>
import { mapState, mapActions } from 'vuex'
import TaskListItem from '@/components/TaskListItem'
export default {
  name: 'task-list',

  components: {
    TaskListItem
  },

  created () {
    this.getTasks({ first: 20 })
  },

  computed: mapState({
    tasks: state => state.tasks.edges,
    newTaskDescription: state => state.newTask.description
  }),

  methods: {
    ...mapActions({
      getTasks: 'tasks',
      createTask: 'createTask'
    }),

    updateNewTaskDescription (e) {
      this.$store.dispatch('updateNewTaskDescription', e.target.value)
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1, h2 {
  font-weight: normal;
}
ul {
  list-style-type: none;
  padding: 0;
}
</style>
