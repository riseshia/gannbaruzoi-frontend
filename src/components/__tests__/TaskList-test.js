import Vue from 'vue'
import Vuex from 'vuex'
import { shallow } from 'vue-test-utils'
import TaskList from '@/components/TaskList'
import storeConfig from '@/store/index'

Vue.config.productionTip = false
Vue.use(Vuex)

const store = new Vuex.Store(storeConfig)

describe('TaskList.vue', () => {
  const wrapper = shallow(TaskList, { store })
  beforeEach(() => {
    store.dispatch = jest.fn()
  })
  it('renders correct contents', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('getTasks', () => {
    wrapper.vm.getTasks({ first: 20 })
    expect(store.dispatch.mock.calls).toEqual([['TASKS', { first: 20 }]])
  })
  it('createTask', () => {
    wrapper.vm.createTask()
    expect(store.dispatch.mock.calls).toEqual([['CREATE_TASK']])
  })
  it('updateNewTaskDescription', () => {
    wrapper.vm.updateNewTaskDescription({ target: { value: '1' } })
    expect(store.dispatch.mock.calls).toEqual([
      ['UPDATE_NEW_TASK_DESCRIPTION', '1']
    ])
  })
})
