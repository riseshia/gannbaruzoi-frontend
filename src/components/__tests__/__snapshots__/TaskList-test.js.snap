// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList.vue renders correct contents 1`] = `
<div class="task-list">
    <h2>Task List</h2> <input type="text" placeholder="Type new task!">
    <ul></ul>
</div>
`;
